src/order_backup.cpp

```diff
@@ -83,7 +84,7 @@ void OrderBackup::DoRestore(Vehicle *v)
 		InvalidateWindowClassesData(WC_STATION_LIST, 0);
 	}
 
-	v->CopyConsistPropertiesFrom(this);
+	if (IsUniqueVehicleName(this->name)) v->CopyConsistPropertiesFrom(this);
```

Comment:
I would expect only the name to not be restored, not all the other flags. Maybe a parameter whether the name must be copied defaulted to `true`?

Reply:
Oops. Messed this up didn't I :/